lib/workers/repository/extract/manager-files.ts

```diff
@@ -1,18 +1,61 @@
 import is from '@sindresorhus/is';
-import type { WorkerExtractConfig } from '../../../config/types';
+import type {
+  CustomManager,
+  ManagerConfig,
+  MatchStringsStrategy,
+} from '../../../config/types';
 import { logger } from '../../../logger';
 import {
   extractAllPackageFiles,
   extractPackageFile,
   get,
 } from '../../../modules/manager';
-import type { PackageFile } from '../../../modules/manager/types';
+import type {
+  CustomExtractConfig,
+  ExtractConfig,
+  PackageFile,
+} from '../../../modules/manager/types';
 import { readLocalFile } from '../../../util/fs';
 
+export function getExtractConfig(config: ManagerConfig): ExtractConfig {
+  const result: ExtractConfig = {
+    npmrc: config.npmrc,
+    aliases: config.aliases,
+    skipInstalls: config.skipInstalls,
+    npmrcMerge: config.npmrcMerge,
+  };
+  return result;
+}
+
+export function getCustomExtractConfig(
+  config: ManagerConfig & CustomManager
+): CustomExtractConfig {
+  const result: CustomExtractConfig = {
+    matchStrings: config.matchStrings,
+    autoReplaceStringTemplate: config.autoReplaceStringTemplate,
+    matchStringsStrategy: config.matchStringsStrategy as MatchStringsStrategy,
+    depNameTemplate: config.depNameTemplate,
+    packageNameTemplate: config.packageNameTemplate,
+    datasourceTemplate: config.datasourceTemplate,
+    versioningTemplate: config.versioningTemplate,
+    depTypeTemplate: config.depTypeTemplate,
+    currentValueTemplate: config.currentValueTemplate,
+    currentDigestTemplate: config.currentDigestTemplate,
+    extractVersionTemplate: config.extractVersionTemplate,
+    registryUrlTemplate: config.registryUrlTemplate,
+  };
+
+  return result;
+}
+
 export async function getManagerPackageFiles(
-  config: WorkerExtractConfig
+  config: ManagerConfig | (ManagerConfig & CustomManager)
```

Comment:
Do you think there's much benefit to split these types and have separate functions, instead of sharing a common type?

Reply:
good idea, let's split the types

Reply:
`ManagerConfig` is introduced to handle manager-config only so its better to integrate `CustomManager` fields inside it as well. More logical that way.
So we can do this:
```ts
 interface ManagerConfig extends RenovateConfig, CustomManager {
 manager: string;
  language?: string | null;
  }

Reply:
Actually I as wondering if we should be *combining* them too. e.g. put all fields in `ManagerConfig` as one possibility. Can you clarify on your suggestion? The above approach seems to complex to me. At least I'd prefer that the Custom/regex manager extends ManagerConfig if it shares fields.

Reply:
I think separate functions will still be needed since we are feeding diff configs based on diff extraction logic, that said main reason to have diff fns was to have only necessary fields in extractConfig when computing fingerprint.

Reply:
I don't think we should see the other fields in non-regex managers because they are only configurable within the `regexManagers` list. And if we see them, they're only default empty values, so it shouldn't affect our fingerprinting.

Reply:
Also the first comment you made can be taken 2 ways:
1. in-favour of splitting types 2. against it 
I am not sure which one you meant. I am answering based on my reasons to make the modifications.

The goal of above splitting `ManagerConfig | (ManagerConfig & CustomManager)` is to ensure consistent typing and prevent future confusion.  So a combined manager config class is a good refactor.
> Before it was of type: `WorkerExtractConfig` but the actual object being passed is of type: `ManagerConfig & 
    Partial<CustomManager>`

Won't adding `ManagerConfig` fields to `Custom/RegexManager` make it bulkier with no actual use-case?

Clarification part:
My suggestion is to just add all fields(of CustomManager) to `ManagerConfig` so the possible changes will be:
1. above [comment](https://github.com/renovatebot/renovate/pull/15919#discussion_r894129391)
2. ```function getManagerPackageFiles(config: ManagerConfig)```
No major benefits except clean code & consistent typing.

Reply:
I think we can extract a generic function from this, so we only have a simple if / else here.
That would make code and types more clean. We can then extend the regex manager config from manager config type.